internal/clientconn/conn.go

```diff
@@ -327,78 +349,54 @@ func (c *conn) processMessage(ctx context.Context, bufr *bufio.Reader, bufw *buf
 // They also should not use recover(). That allows us to use fuzzing.
 //
 // Returned resBody can be nil.
-func (c *conn) route(connCtx context.Context, reqHeader *wire.MsgHeader, reqBody wire.MsgBody) (resHeader *wire.MsgHeader, resBody wire.MsgBody, closeConn bool) { //nolint:lll // argument list is too long
-	var span oteltrace.Span
+func (c *conn) route(connCtx context.Context, reqHeader *wire.MsgHeader, reqBody wire.MsgBody, command string) (resHeader *wire.MsgHeader, resBody wire.MsgBody, closeConn bool) { //nolint:lll // argument list is too long
+	span := oteltrace.SpanFromContext(connCtx)
 
-	var command, result, argument string
+	var result, argument, metricsCommand string
 	defer func() {
-		if result == "" {
-			result = "panic"
-		}
-
 		if argument == "" {
 			argument = "unknown"
 		}
 
-		c.m.Responses.WithLabelValues(resHeader.OpCode.String(), command, argument, result).Inc()
+		setSpanAttribute(span, result, argument)
 
-		must.NotBeZero(span)
+		// extract metrics out of this function, currently it requires labels based on error
+		// TODO https://github.com/FerretDB/FerretDB/issues/1997
+		c.m.Responses.WithLabelValues(resHeader.OpCode.String(), metricsCommand, argument, result).Inc()
```

Comment:
… here

Reply:
Thanks, yes I see where this was done wrong.